packages/editor/src/components/post-author/index.js

```diff
@@ -1,71 +1,137 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
 /**
  * WordPress dependencies
  */
+import { useState, useMemo } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { withInstanceId, compose } from '@wordpress/compose';
-import { Component } from '@wordpress/element';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { decodeEntities } from '@wordpress/html-entities';
+import { ComboboxControl } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import PostAuthorCheck from './check';
 
-export class PostAuthor extends Component {
-	constructor() {
-		super( ...arguments );
+let latestAvailableAuthors;
+function PostAuthor() {
+	const authors = useSelect( ( select ) => select( 'core' ).getAuthors() );
+	const postAuthor = useSelect( ( select ) =>
+		select( 'core' ).getAuthor(
+			select( 'core/editor' ).getEditedPostAttribute( 'author' )
+		)
```

Comment:
There is no need for two `useSelect` calls here.

Reply:
really, i thought this was the pattern I saw elsewhere in GB. we need two separate values - the authors and the post author. 

so instead I should to a single useSelect that selects and returns both them restructure them?

Reply:
@gwwar That makes sense, thank you for the code sample which helped me understand the request. Used slightly modified to resolve in [`66f023c` (#23237)](https://github.com/WordPress/gutenberg/pull/23237/commits/66f023c2daba60aebd793e6de8f8eb5874f4e0cf).